### DISCUSSION TITLE
Rerunning an expected crontab for a given date

### DISCUSSION THREAD

**Discussion by austinkregel** [laravel/framework#43007] [Ideas]
Hello! 

**TL;DR;** Would being able to re-run a crontab/defined schedule be of benefit to your usage?

My team has encountered the need, more than once, to rerun a whole day or week's worth of cron jobs (either for data fetching, report building, or data manipulation). I've been attempting to implement the functionality to rerun a cron schedule, and I have the whole schedule working. However, we quickly ran into the issue of not needing to run the _entire_ cron, but only parts of the cron. Which I think gave me a decent idea.

Cron scheduler tags/labels/groups. Or otherwise associating a string with a given cron event, and being able to choose to only re-run crons with that same string.

Essentially, the whole idea of the feature is based around this little bit of code to rerun a cron schedule as if it were running in real time.

```php
Artisan::command('schedule:rerun {date} {--tags=}', function () {
    $date = $this->argument('date');

    $parsedDate = Carbon::parse($date);
    $endOfDay = $parsedDate->copy()->endOfDay()->addSecond();

    while ($parsedDate->isBefore($endOfDay)) {
        Carbon::setTestNow($parsedDate);
        Artisan::call('schedule:run', [
            '--tags' => $this->option('tags'),
        ], $this->output);

        $parsedDate->addMinute();
    }
});
```

The part that seems to be impractical as a package, is adding the tags method to the framework so you could use a schedule like this

```php
protected function schedule(Schedule $schedule): void
{
     $schedule->job(Jobs\Google\SyncAnalyticsProfilesJob::class)->hourly()->tags('google', 'data');
     $schedule->job(Jobs\Facebook\RefreshAssetsJob::class)->hourly()->tags('facebook', 'data');
     $schedule->job(Jobs\Insights\InsightCachingJob::class)->dailyAt('01:00')->tags('internal', 'data');
     $schedule->job(Jobs\Insights\CheckForEnabledCampaignsWithPausedLabelsJob::class)->dailyAt('02:00')->tags('data');
     $schedule->job(Jobs\Facebook\RefreshStatsJob::class)->monthlyOn(28, '15:00')->tags('facebook', 'data');
     /// ...etc
}
```
I'm able to extend the code to the point of being able to use the above code locally, however creating my own version of `schedule:run` command results in the schedule being empty (or otherwise not bound to my new schedule run command). However, [modifying the existing `schedule:run` command at this line](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Console/Scheduling/ScheduleRunCommand.php#L111) with the below code will do exactly as I'm expecting and launch all the jobs that are tagged with the provided tag.

```php
    if ($this->hasOption('tags') && !property_exists($event, 'tags')) {
          $this->dispatcher->dispatch(new ScheduledTaskSkipped($event));
          continue;
    }

    if ($this->hasOption('tags') && property_exists($event, 'tags')) {
        $tags = explode(',', $this->option('tags'));


        if (empty(array_intersect($tags, $event->tags))) {
            $this->dispatcher->dispatch(new ScheduledTaskSkipped($event));

            continue;
        }
    }
```

I'm adding a small enough amount of code for the overall feature that I'd happily make a PR to fully flush this out if there's interest.
